alf/algorithms/td_loss.py

```diff
@@ -99,15 +102,37 @@ def forward(self, experience, value, target_value):
                 values=target_value,
                 step_types=experience.step_type,
                 discounts=experience.discount * self._gamma)
-        else:
+        elif train_info == None:
```

Comment:
Instead of checking whether train_info is None, you should add an argument in `__init__` to indicate whether use retrace.
You should also change SarsaAlgorithm and SacAlgorithm to pass in train_info.